apps/rush-lib/src/logic/InstallManager.ts

```diff
@@ -1029,6 +1064,49 @@ export class InstallManager {
     });
   }
 
+  private _mergeEnvironmentVariables(
+    baseEnv: NodeJS.ProcessEnv,
+    environmentVariables?: IConfigurationEnvironment
+  ): NodeJS.ProcessEnv {
+    const packageManagerEnv: NodeJS.ProcessEnv = baseEnv;
+
+    if (environmentVariables) {
+      // eslint-disable-next-line guard-for-in
+      for (const envVar in environmentVariables) {
+        let setEnvironmentVariable: boolean = true;
+        console.log(`\nProcessing definition for environment variable: ${envVar}`);
+
+        if (process.env[envVar]) {
+          setEnvironmentVariable = false;
+          console.log(colors.yellow(`WARNING: Environment variable already defined:`));
+          console.log(`  Name: ${envVar}`);
+          console.log(`  Value set on the device: ${process.env[envVar]}`);
+          console.log(`  Value set in rush.json: ${environmentVariables[envVar].value}`);
+
+          if (environmentVariables[envVar].override &&
+            environmentVariables[envVar].override === true) {
+            setEnvironmentVariable = true;
+            console.log(colors.yellow(`WARNING: Overriding the environment variable with the value set in rush.json.`));
```

Comment:
Now that I'm thinking about this again, should we bother warning at all when we override the environment variable? Maybe we should just log that we're overriding an existing variable. The purpose of the `override` option is to let the repo owner control which ones should be overridden, so they should already be aware of what will happen.

Reply:
Makes sense.